make/hotspot/lib/JvmDtraceObjects.gmk

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright (c) 2013, 2019, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.  Oracle designates this
+# particular file as subject to the "Classpath" exception as provided
+# by Oracle in the LICENSE file that accompanied this code.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+ifeq ($(call check-jvm-feature, dtrace), true)
+  ifeq ($(OPENJDK_TARGET_OS), bsd)
+
+    ############################################################################
+    # Integrate with libjvm. Here we generate an object file which is
+    # linked with libjvm.so. This step is complicated from a dependency
+    # perspective. We add this file to the linking of libjvm using
+    # EXTRA_OBJECT_FILES, but it needs to be created outside the call to
+    # SetupNativeCompilation. Also, this files is dependent on compiled
+    # object files from the libjvm compilation, so this generation must happen
+    # as a part of the libjvm compilation.
+
+    DTRACE_OBJ := $(JVM_OUTPUTDIR)/objs/dtrace.o
```

Comment:
I just want to double-check before accepting this piece of necromancy -- is dtrace really used on BSD, in a way that this instrumentation is needed? There is some dtrace support already in the system, in the form of generated header files in make/hotspot/gensrc/GensrcDtrace.gmk. Maybe that is enough to achieve what you want?

(I will admit that I don't really understand what this dtrace instrumentation does, or how it differs from the generated code we still keep. What I do know is that this is kludgy from a built perspective, and I'd like to keep it in its grave, if possible.)